node/src/types/block.rs

```diff
@@ -1125,7 +1128,7 @@ pub(crate) mod json_compatibility {
         hash: BlockHash,
         header: JsonBlockHeader,
         body: (),
-        proofs: Vec<Signature>,
+        proofs: BTreeMap<PublicKey, Signature>,
```

Comment:
Unfortunately we need to avoid using `BTreeMap`s in the JSON representations.  The problem is that serde_json encodes each entry as a separate, unique JSON object, which means we can't define a schema for it.

We need to convert this to a `Vec<JsonProof>` here so it becomes a JSON array type, where `JsonProof` is something like:
```
/// A Json representation of a single bid.
#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, JsonSchema)]
#[serde(deny_unknown_fields)]
pub struct JsonProof {
    public_key: PublicKey,
    signature: Signature,
}
```

I haven't thought of a good way to catch this so far - it's an easy thing to forget.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/656/commits/db26360a453cdef92fea0ef2499329bd403899e6.